api/v2/checluster_types.go

```diff
@@ -0,0 +1,639 @@
+	// See the doc https://docs.openshift.com/container-platform/4.4/networking/enable-cluster-wide-proxy.html. See also the `proxyURL` fields.
+	NonProxyHosts []string `json:"nonProxyHosts,omitempty"`
+	// The secret name that contains `user` and `password` for a proxy server.
+	// The secret must have `app.kubernetes.io/part-of=che.eclipse.org` label.
+	// +optional
+	CredentialsSecretName string `json:"credentialsSecretName,omitempty"`
+}
+
+// PersistentVolumeClaim custom settings.
+type PVC struct {
+	// Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.
+	// +optional
+	ClaimSize string `json:"claimSize,omitempty"`
+	// Storage class for the Persistent Volume Claim. When omitted or left blank, a default storage class is used.
+	// +optional
+	StorageClass string `json:"storageClass,omitempty"`
+}
+
+// External devfile registries configuration.
+type ExternalDevfileRegistry struct {
+	// Public URL of the devfile registry that serves sample ready-to-use devfiles.
+	// +optional
+	Url string `json:"url,omitempty"`
+}
+
+// External plugin registries configuration.
+type ExternalPluginRegistry struct {
+	// Public URL of the plugin registry.
+	// +optional
+	Url string `json:"url,omitempty"`
+}
+
+// Deployment custom settings.
+type Deployment struct {
+	// A single application container.
+	// +optional
+	Containers []Container `json:"container,omitempty"`
+	// Security options the pod should run with.
+	// +optional
+	SecurityContext PodSecurityContext `json:"securityContext,omitempty"`
+}
+
+// Container custom settings.
+type Container struct {
+	// Container name.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Container image. Omit it or leave it empty to use the default container image provided by the Operator.
+	// +optional
+	Image string `json:"image,omitempty"`
+	// Image pull policy. Default value is `Always` for `nightly`, `next` or `latest` images, and `IfNotPresent` in other cases.
+	// +optional
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+	// Compute Resources required by this container.
+	// +optional
+	Resources ResourceRequirements `json:"resources,omitempty"`
+}
+
+// Describes the compute resource requirements.
+type ResourceRequirements struct {
+	// Requests describes the minimum amount of compute resources required.
+	// +optional
+	Requests ResourceList `json:"request,omitempty"`
+	// Limits describes the maximum amount of compute resources allowed.
+	// +optional
+	Limits ResourceList `json:"limits,omitempty"`
+}
+
+// List of resources.
+type ResourceList struct {
+	// Memory, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
+	// +optional
+	Memory string `json:"memory,omitempty"`
+	// CPU, in cores. (500m = .5 cores)
+	// +optional
+	Cpu string `json:"cpu,omitempty"`
```

Comment:
These could be specified as `resource.Quantity` to get the standard k8s validation automatically and avoid the need to parse them in code.

```go
import "k8s.io/apimachinery/pkg/api/resource"
//...
	// Memory, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
	// +optional
	Memory resource.Quantity `json:"memory,omitempty"`
	// CPU, in cores. (500m = .5 cores)
	// +optional
	Cpu resource.Quantity `json:"cpu,omitempty"`
```